annotation_store/store.py

```diff
@@ -0,0 +1,542 @@
+        self.grade_passback_outcome = None
+        self.logger = logging.getLogger('{module}.{cls}'.format(module=__name__, cls=self.__class__.__name__))
+        assert self.backend is not None
+        assert isinstance(self.gather_statistics, bool)
+
+    @classmethod
+    def from_settings(cls, request):
+        gather_statistics = cls.SETTINGS.get('gather_statistics', False)
+        backend_type = cls.SETTINGS.get('backend', 'catch')
+        backend_types = {'app': AppStoreBackend, 'catch': CatchStoreBackend}
+        assert backend_type in backend_types
+        backend_instance = backend_types[backend_type](request)
+        return cls(request, backend_instance=backend_instance, gather_statistics=gather_statistics)
+
+    @classmethod
+    def update_settings(cls, settings_dict):
+        cls.SETTINGS = settings_dict
+        return cls
+
+    def root(self):
+        return self.backend.root()
+
+    def index(self):
+        raise NotImplementedError
+
+    def search(self):
+        self.logger.info(u"Search: %s" % self.request.GET)
+        self._verify_course(self.request.GET.get('contextId', None))
+        if hasattr(self.backend, 'before_search'):
+            self.backend.before_search()
+        return self.backend.search()
+
+    def create(self):
+        body = json.loads(self.request.body)
+        self.logger.info(u"Create annotation: %s" % body)
+        self._verify_course(body.get('contextId', None))
+        self._verify_user(body.get('user', {}).get('id', None))
+        if hasattr(self.backend, 'before_create'):
+            self.backend.before_create()
+        response = self.backend.create()
+        self.after_create(response)
+        return response
+
+    def after_create(self, response):
+        is_graded = self.request.LTI.get('is_graded', False)
+        self._lti_grade_passback(is_graded=is_graded, status_code=response.status_code, result_score=1)
+        self._update_stats('create', response)
+
+    def read(self, annotation_id):
+        raise NotImplementedError
+
+    def after_read(self, response):
+        self._update_stats('read', response)
+
+    def update(self, annotation_id):
+        body = json.loads(self.request.body)
+        self.logger.info(u"Update annotation %s: %s" % (annotation_id, body))
+        self._verify_course(body.get('contextId', None))
+        self._verify_user(body.get('user', {}).get('id', None))
+        if hasattr(self.backend, 'before_update'):
+            self.backend.before_update(annotation_id)
+        response = self.backend.update(annotation_id)
+        self.after_update(response)
+        return response
+
+    def after_update(self, response):
+        self._update_stats('update', response)
+
+    def delete(self, annotation_id):
+        body = json.loads(self.request.body)
+        self.logger.info(u"Delete annotation %s: %s" % (annotation_id, body))
+        self._verify_course(body.get('contextId', None))
```

Comment:
Does your version of Mirador send the annotation in the body? Mine does not so it fails this verification and doesn't actually delete.

Reply:
No you're right, Mirador just sends the ID. Fixed in 7c936d295c09819bb874b126969c0bc5aeb65c47.